File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:

        stack = []

        for i in tokens:
            if i != "+" and i != "-" and i != "*" and i!= "/":
                stack.append(int(i))
            else:
                secondNum = stack.pop()
                firstNum = stack.pop()

                if i == "+":
                    stack.append(firstNum + secondNum)
                elif i == "-":
                    stack.append(firstNum - secondNum)
                elif i == "/":
                    stack.append(int(firstNum/secondNum))
                elif i == "*":
                    stack.append(firstNum*secondNum)
                
        return stack.pop()
        

File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.java
class Solution {
    public int evalRPN(String[] tokens) {
        
        Stack<Integer> st = new Stack<>();

        for(int i=0;i<tokens.length;i++)
        {
            if( !st.empty() && tokens[i].equals("/") )
            {
                int n2 = st.peek();
                st.pop();
                int n1 = st.peek();
                st.pop();
                st.push(n1/n2);
            }
            else if( !st.empty() && tokens[i].equals("+") )
            {
                int n2 = st.peek();
                st.pop();
                int n1 = st.peek();
                st.pop();
                st.push(n1+n2);
            }
            else if( !st.empty() && tokens[i].equals("*") )
            {
                int n2 = st.peek();
                st.pop();
                int n1 = st.peek();
                st.pop();
                st.push(n1*n2);
            }
            else if( !st.empty() && tokens[i].equals("-") )
            {
                int n2 = st.peek();
                st.pop();
                int n1 = st.peek();
                st.pop();
                st.push(n1-n2);
            }
            else
            {
                st.push( Integer.parseInt(tokens[i]) );
            }
        }

        return st.peek(); 
    }
}